examples/gio_dbus_register_object/main.rs

```diff
@@ -22,25 +58,24 @@ fn on_startup(app: &gio::Application, tx: &Sender<gio::RegistrationId>) {
 
     if let Ok(id) = connection
         .register_object("/com/github/gtk_rs/examples/HelloWorld", &example)
-        .method_call(glib::clone!(
-            #[strong]
-            app,
-            move |_, _, _, _, method, params, invocation| {
-                match method {
-                    "Hello" => {
-                        if let Some((name,)) = <(String,)>::from_variant(&params) {
-                            let greet = format!("Hello {name}!");
-                            println!("{greet}");
-                            invocation.return_value(Some(&(greet,).to_variant()));
-                        } else {
-                            invocation.return_error(gio::IOErrorEnum::Failed, "Invalid parameters");
-                        }
+        .method_call(move |_, _, _, _, method, params, invocation| {
```

Comment:
I think this could be improved quite a bit by introducing a new trait that would implement the parsing and you could just call method_call::<T> and it would pass you the parsed method already

Reply:
Sounds like a good idea. I'll try to add this 🙏

Reply:
@bilelmoussaoui I've pushed a new commit which provides an API around handling "parsed" calls.

The builder interface became a bit more intricate to "hide" the closure type parameter, but I think that's an acceptable price for the added convenience :innocent: 

I've also added overloads which hide the `invocation` object and instead return a result from the closure, which is a lot safer, as you can't forget to call `invocation.return_*` anymore :slightly_smiling_face:

Reply:
That looks like an improvement indeed. A derive macro for this would also be relatively simple to write.
I'm not sure about the method name (`parse_method()` at that place seems a bit out of place).

Reply:
Yeah, the name 😬 Just couldn't come up with any better 🤔

Would `build_method_call` be a better choice?

Reply:
As for the derive macro, I'm not sure I'd be able to write this any time soon.  I've never done this before, see my other comment :)